src/Dofs/ConstraintHandler.jl

```diff
@@ -306,7 +306,7 @@ function add_prescribed_dof!(ch::ConstraintHandler, constrained_dof::Int, inhomo
     return ch
 end
 
-function _add!(ch::ConstraintHandler, dbc::Dirichlet, bcfaces::Set{Index}, interpolation::Interpolation, field_dim::Int, offset::Int, bcvalue::BCValues, cellset::Set{Int}=Set{Int}(1:getncells(ch.dh.grid))) where {Index<:BoundaryIndex}
+function _add!(ch::ConstraintHandler, dbc::Dirichlet, bcfaces::Set{Index}, interpolation::Interpolation, field_dim::Int, offset::Int, bcvalue::BCValues, cellset=1:getncells(ch.dh.grid)) where {Index<:BoundaryIndex}
```

Comment:
Here it might be nice to keep since it is used on L319 to check `cellidx ∉ cellset`. Still fast with a `UnitRange`, but if you pass a vector it will be slow(er).

Reply:
Some rough benchmarks seem to suggest that it is benefitial to create the set if `length(needles) > length(haystack)`, i.e. if we the number of lookups we will do is larger than the size of the collection. Still pretty fast though so perhaps  just leave it as a vector if one passes that. In most cases it will already be a set, since it will come from a grid-cellset, which are sets.

Reply:
I also thought about this, but decided to go for creating the Set always. We gain significant performance benefits if the Vector is large, while we definitely get some overhead due to the Set construction. However, for small vectors (where it may be enough to not convert it to a Set upfront), the overhead is rather small, so searching for a good heuristic might not be really worth here. I would still go for the conversion to Set, as the speedup can be really significant for large problems (O(N) vs O(log(N)).